utils/misc.mli

```diff
@@ -314,3 +314,26 @@ val normalise_eol : string -> string
 val delete_eol_spaces : string -> string
 (** [delete_eol_spaces s] returns a fresh copy of [s] with any end of line spaces
    removed. Intended to normalize the output of the toplevel for tests. *)
+
+
+
+
+(* Hooks machinery:
+  [add_hook name f] will register a function that will be called on the
+    argument of a later call to [apply_hooks]. Hooks are applied in the
+    lexicographical order of their names.
+*)
+
+exception HookExn of exn
```

Comment:
The idea is that a hook raising an exception is almost always an error, in which case it should be detected and caught immediately. However, in our experience, there are cases where you really want to raise an exception, that should go outside of the hook. Raising `HookExn exn` is thus the correct way to raise an exception outside of the hook. For the error, I could indeed create an exception for that case, but in `Misc`, there is currently no support for printing errors in the standard way.